.github/workflows/ci.yml

```diff
@@ -0,0 +1,37 @@
+---
+name: Continuous Integration
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  contents: read
+
+jobs:
+  embedded_build:
+    name: Embedded Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        gcc: ["8-2019-q3", "9-2020-q2", "10.3-2021.10"]
+        configuration: ["RelWithDebInfo"]
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      - name: Install GNU Arm Embedded Toolchain ${{ matrix.gcc }}
+        uses: carlosperate/arm-none-eabi-gcc-action@828dd7ce6ea122338b9bafd5d9969de1b5dfbfe2
+        with:
+          release: ${{ matrix.gcc }}
+      - uses: seanmiddleditch/gha-setup-ninja@16b940825621068d98711680b6c3ff92201f8fc0
+      - uses: hendrikmuhs/ccache-action@ed038da2f2f09b0c8387c00e1498290a4808da2e
```

Comment:
It is advised to pin to commit-hashes instead of versions. Versions are tags and can be moved or manipulated. A commit hash uniquely pins the version. And Dependabot is responsible for updating the dependencies; so no need for readable versions here.